hpx/runtime/serialization/detail/pointer.hpp

```diff
@@ -176,7 +177,26 @@ namespace hpx { namespace serialization
             {
                 static void call(output_archive& ar, const Pointer& ptr)
                 {
-                    ar << *ptr;
+                    call(ar, *ptr);
+                }
+
+                template <typename T>
+                static typename std::enable_if<
+                    !std::is_constructible<T>::value
+                >::type
+                call(output_archive& ar, T const& val)
+                {
+                    save_construct_data(ar, &val, 0);
```

Comment:
See above. The reasons are mostly symmetry (if `load_construct_data` expects to read stuff from the archive this should be written in `save_construct_data`) and compatibility with  Boost serialization where this technique is used for serializing containers and pointers of/to non-default-constructible types.